### ISSUE TITLE
UnityWebRequestMultimedia.GetAudioClip error with Forget

### ISSUE THREAD

**Issue by kamyker** [Cysharp/UniTask#171]
Doing:
`RefreshHitAudioClip().Forget();`

```
public static async UniTask RefreshHitAudioClip()
{
	try
	{
		string hitAudioClipPath = "hitSound.ogg";
		if ( File.Exists( hitAudioClipPath ) )
		{
			using ( var www = UnityWebRequestMultimedia.GetAudioClip( hitAudioClipPath, AudioType.OGGVORBIS ) )
			{
				await www.SendWebRequest();//.ToUniTask(); works fine with ToUniTask
				Loaded.HitAudioClip = DownloadHandlerAudioClip.GetContent( www );
			}
		}
	}
	catch ( Exception e )
	{
		Debug.LogException( e );
	}
}
```
Gives error:
```
InvalidOperationException: Token version is not matched, can not await twice or get Status after await.
Cysharp.Threading.Tasks.UniTaskCompletionSourceCore`1[TResult].ValidateToken (System.Int16 token) (at Library/PackageCache/com.cysharp.unitask@2bf9f4f062/Runtime/UniTaskCompletionSource.cs:304)
Cysharp.Threading.Tasks.UniTaskCompletionSourceCore`1[TResult].OnCompleted (System.Action`1[T] continuation, System.Object state, System.Int16 token) (at Library/PackageCache/com.cysharp.unitask@2bf9f4f062/Runtime/UniTaskCompletionSource.cs:265)
Cysharp.Threading.Tasks.CompilerServices.AsyncUniTask`1[TStateMachine].OnCompleted (System.Action`1[T] continuation, System.Object state, System.Int16 token) (at Library/PackageCache/com.cysharp.unitask@2bf9f4f062/Runtime/CompilerServices/StateMachineRunner.cs:246)
Cysharp.Threading.Tasks.UniTask+Awaiter.SourceOnCompleted (System.Action`1[T] continuation, System.Object state) (at Library/PackageCache/com.cysharp.unitask@2bf9f4f062/Runtime/UniTask.cs:351)
Cysharp.Threading.Tasks.UniTaskExtensions.Forget (Cysharp.Threading.Tasks.UniTask task) (at Library/PackageCache/com.cysharp.unitask@2bf9f4f062/Runtime/UniTaskExtensions.cs:565)
```

As shown above `await www.SendWebRequest().ToUniTask();` works fine

**Comment by neuecc** [comment]
thanks, but I can't reproduce it.
and is your stacktrace is right?
it seems that indicated message is not about unitywebrequest.getawaiter(await direct).

**Comment by kamyker** [comment]
> it seems that indicated message is not about unitywebrequest.getawaiter(await direct)
It's about Forget + that await somehow.
![](https://i.gyazo.com/7b38075b709983490f0ae086afbb03d9.png)

**Comment by kamyker** [comment]
Somehow I got this error only once per game run, Reset runs too fast?

![](https://i.imgur.com/Bsf2hbX.gif)

**Comment by kamyker** [comment]
This fixes my issue:

Changed:
```
        public static void Forget(this UniTask task)
        {
            var awaiter = task.GetAwaiter();
            if (awaiter.IsCompleted)
            {
                try
                {
                    awaiter.GetResult();
                }
                catch (Exception ex)
                {
                    UniTaskScheduler.PublishUnobservedTaskException(ex);
                }
            }

            awaiter.SourceOnCompleted(state =>
            {
                using (var t = (StateTuple<UniTask.Awaiter>)state)
                {
                    try
                    {
                        t.Item1.GetResult();
                    }
                    catch (Exception ex)
                    {
                        UniTaskScheduler.PublishUnobservedTaskException(ex);
                    }
                }
            }, StateTuple.Create(awaiter));
        }
```

To:
```
        public static void Forget(this UniTask task)
        {
            var awaiter = task.GetAwaiter();
            if (awaiter.IsCompleted)
            {
                try
                {
                    awaiter.GetResult();
                }
                catch (Exception ex)
                {
                    UniTaskScheduler.PublishUnobservedTaskException(ex);
                }
            }
            else
            {
                awaiter.SourceOnCompleted(state =>
                {
                    using (var t = (StateTuple<UniTask.Awaiter>)state)
                    {
                        try
                        {
                            t.Item1.GetResult();
                        }
                        catch (Exception ex)
                        {
                            UniTaskScheduler.PublishUnobservedTaskException(ex);
                        }
                    }
                }, StateTuple.Create(awaiter));
            }
        }
```

Not sure how UniTask works exactly but this makes sense otherwise GetResult() runs twice when awaiter.IsCompleted == true.

**Comment by neuecc** [comment]
Sorry, I missed that Forget was the problem.
And thank you for the details, you're right!
I'll fix and release it soon.

**Comment by neuecc** [comment]
Fixed at `2.0.36`.
